Makefile

```diff
@@ -0,0 +1,154 @@
+_THIS    := $(realpath $(dir $(abspath $(lastword $(MAKEFILE_LIST)))))
+_SRC     := $(_THIS)/src_files
+_BUILD   := $(_THIS)/_build
+_BIN     := $(_THIS)/bin
+_MAKE    := $(MAKE) --no-print-directory -C $(_THIS)
+
+CC        = gcc
+CXX       = g++
+_CSRCS   := $(_SRC)/syzygy/tbprobe.c
+_CXXSRCS := $(wildcard $(_SRC)/*.cpp)
+_SRCS    := $(_CSRCS) $(_CXXSRCS)
+_COBJS   := $(patsubst $(_SRC)/%.c,$(_BUILD)/%.o,$(_CSRCS))
+_CXXOBJS := $(patsubst $(_SRC)/%.cpp,$(_BUILD)/%.o,$(_CXXSRCS))
+_OBJS    := $(_COBJS) $(_CXXOBJS)
+
+NAME      = Koivisto
+MINOR     = 9
+MAJOR     = 4
+_TARGET  := $(_BIN)/$(NAME)_$(MAJOR).$(MINOR)
+
+
+ifeq ($(OS),Windows_NT)
+PREFIX    = windows
+SUFFIX    = .exe
+else
+
+UNAME    := $(shell uname -s)
+ifeq ($(UNAME),Linux)
+PREFIX    = linux
+SUFFIX    =
+else
+$(warning incompatible host, this might fail)
+PREFIX    = _
+SUFFIX    =
+endif
+endif
+
+
+# MARK: Args
+
+DEBUG    ?= 1
+POPCNT   ?= 1
+PEXT     ?= 0
+AVX      ?= 0
+MODE     ?= native
+
+ifeq ($(MODE),release)
+DEBUG     = 0
+endif
+
+
+# MARK: Flags
+
+_MACROS   = -DMINOR_VERSION=$(MINOR) -DMAJOR_VERSION=$(MAJOR)
+_WARN     = -Wall -Wextra -Wshadow
+_CFLAGS   = $(_WARN) $(_MACROS)
+_CXXFLAGS = -std=c++17
+_LDFLAGS  = -pthread -Wl,--whole-archive -lpthread -Wl,--no-whole-archive
```

Comment:
Also, I'm not entirely sure whether the `-pthread` flag should also be used during compilation, since currently it's only used at the linking stage. Could add it to `_CFLAGS` to be sure.

Reply:
This is the kind of thing is where we just add a bunch of flags just to be sure it works, but obviously a proper solution is better :)

Reply:
Yeah, if I understood the GCC docs correctly the `-pthread` flag should probably be added to the compilation as well 🙂